client/lib/cart-values/cart-items.ts

```diff
@@ -165,6 +165,10 @@ export function hasTransferProduct( cart: ObjectWithProducts ): boolean {
 	return getAllCartItems( cart ).some( isDomainTransfer );
 }
 
+export function hasTransferProductOnly( cart: ObjectWithProducts ): boolean {
+	return getAllCartItems( cart ).every( isDomainTransfer );
+}
+
```

Comment:
Just in case it's relevant, any credits in the cart will show up as a product, which means that this would be falsy even if the cart contains only domain transfers if the user has credits. See `hasOnlyRenewalItems` for an example of maybe a better way to do this.